File: app/src/main/java/com/botijasoftware/vrtest/MainActivity.java
package com.botijasoftware.vrtest;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.os.Bundle;
import com.botijasoftware.utils.GameActivityVR;

public class MainActivity extends GameActivityVR {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mScreenManager.addScreen(new MainScreen(mScreenManager));

    }

    @Override
    protected void onPostCreate(Bundle savedInstanceState) {
        super.onPostCreate(savedInstanceState);

    }

    @Override
    public void onBackPressed() {

        if (!mScreenManager.onBackPressed()) {
            new AlertDialog.Builder(this)
                    .setIcon(android.R.drawable.ic_dialog_alert)
                    .setTitle("Quit game?")
                    .setMessage("Are you sure you want to quit?")
                    .setPositiveButton("Yes", new DialogInterface.OnClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog, int which) {
                            finish();
                        }

                    })
                    .setNegativeButton("No", null)
                    .show();
        }
    }



}

File: app/src/main/java/com/botijasoftware/vrtest/MainActivity.kt
package com.botijasoftware.vrtest

import android.app.AlertDialog
import android.content.DialogInterface
import android.os.Bundle

import com.botijasoftware.utils.GameActivity
import com.botijasoftware.utils.GameActivityVR

class MainActivity : GameActivityVR() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mScreenManager.addScreen(MainScreen(mScreenManager))

    }

    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)

    }

    override fun onBackPressed() {

        if (!mScreenManager.onBackPressed()) {
            AlertDialog.Builder(this)
                    .setIcon(android.R.drawable.ic_dialog_alert)
                    .setTitle("Quit game?")
                    .setMessage("Are you sure you want to quit?")
                    .setPositiveButton("Yes") { dialog, which -> finish() }
                    .setNegativeButton("No", null)
                    .show()
        }
    }


}
